### DISCUSSION TITLE
How to flash top bar/chapter name when a new chapter plays.

### DISCUSSION THREAD

**Discussion by lewdpatriot** [tomasklaen/uosc#478] [Q&A]
I know how to show UI-elements when clicking a button with this
`>         add chapter 1; script-message-to uosc flash-elements timeline`

But I want to flash chapter name when the video naturally enters a new chapter. 

![Screenshot_20230320_230414](https://user-images.githubusercontent.com/63286528/226421670-2de7370b-0940-4c27-8ca1-ae92c569e9fa.png)
![IMG_20230320_230150 (1)](https://user-images.githubusercontent.com/63286528/226422633-95a6e405-6abe-4257-8e72-0f088d44b8f7.jpg)
Either of these is fine.

**Comment by OmiChrome** [comment]
for that part [Metadata-osd](https://github.com/vc-01/metadata-osd)  is better alternative 

i have put it on middle 
![image](https://user-images.githubusercontent.com/108952834/226447787-7b05feea-4334-429d-983a-aa47f3d18c71.png)

```
style_alignment_osd_1=left_justified;toptitle
style_alignment_osd_2=toptitle;midtitle

```

**Reply by OmiChrome** [reply]
yes it `flash` when new chapter is started

**Reply by lewdpatriot** [reply]
Thank you.

**Comment by christoph-heinrich** [comment]
You need to write a script for that and observe the chapter property.

**Comment by uugaabuugaa** [comment]
I'm also looking for a better solution for this issue. I found this on reddit:
```
local function showChapterTitle()
    local chapterTitle = mp.get_property_osd("chapter-metadata/by-key/title")
    mp.osd_message(chapterTitle, 3)
end

mp.observe_property("chapter", nil, showChapterTitle)
```

which kinda works but doesn't show "Chapter(x):" before the chapter title. Also when there's no chapters in vidoes it shows the file title twice on start of playback which is quite undesirable. Changing `mp.osd_message(chapterTitle, 3)` with `mp.command("script-binding uosc/flash-top-bar")` is one way to flash chapter since there's no `uosc/flash-chapter` command to my knowledge but it is very distracting since file playlist number, file title, window controls will show up as well. Would be nice to have a way to just flash uosc's chapter osd in some way.

EDIT: Someone helped me modify the code a bit and it works perfect for me; with `osd-playing-msg=""` in `mpv.conf` it shows only uosc's top bar flash at file-load followed by chapter-title only if there is chapters available in the file.
```
local function showChapterTitle()
    local chapterExists = mp.get_property("chapter")
    local idle = mp.get_property_native("idle-active")
    
    if chapterExists and not idle then
        mp.osd_message('Chapter: '.. mp.command_native({'expand-text', '${chapter}'}), 3)
    end
end
mp.observe_property("chapter", nil, showChapterTitle)
```

**Comment by tomasklaen** [comment]
You can now add this to your options ([5f70143](https://github.com/tomasklaen/uosc/commit/5f701436d11a6a9e19e73c1c246561b083f45bf1)):

```
top_bar_flash_on=chapter
```

**Reply by uugaabuugaa** [reply]
Great. I'll check that out later when I'm free. Thanks.
